src/components/message/message.stories.tsx

```diff
@@ -20,6 +20,25 @@ export const Default: ComponentStory<typeof Message> = () => {
   );
 };
 
+export const ShowCloseIcon: ComponentStory<typeof Message> = () => {
+  const [isOpen, setIsOpen] = useState(true);
+  return (
+    <>
+      {!isOpen && <Button onClick={() => setIsOpen(true)}>Open Message</Button>}
+      <Box width="600px">
+        <Message
+          showCloseIcon={false}
+          open={isOpen}
+          onDismiss={() => setIsOpen(false)}
+        >
+          A longer custom message which now shows the close icon is not rendered
+          and padding is equal on both sides
+        </Message>
+      </Box>
+    </>
+  );
```

Comment:
Given `onDismiss` is only called when the close icon is clicked, by hiding the icon its not possible to hide the message and access the "Open Message" button, so what do you think about this?:

```suggestion
  return (
      <Box width="600px">
        <Message
          showCloseIcon={false}
          open
        >
          A longer custom message which now shows the close icon is not rendered
          and padding is equal on both sides
        </Message>
      </Box>
  );
```

Visually this would achieve the same result but without having the inaccessible button.